File: 2022/07/part1.cs
string text = File.ReadAllText("input.txt");

var x = text.Split("\r\n");

var path = new Stack<string>();
var map = new Dictionary<string, int>();

for (int i = 0; i < x.Length; i++)
{
    var h = x[i].Split(' ');
    if (h[0] == "$")
    {

        if (h[1] == "cd")
        {
            if (h[2] == "..")
            {
                path.Pop();
            }
            else if (h[2] == "/")
            {
                path.Push(string.Join("",path) + h[2]);
            }
            else
            {
                path.Push(string.Join("", path) + h[2]);
            }
        }
    } else if (h[0] != "dir") {

        var size = int.Parse(h[0]);
        foreach (var item in path)
        {
            map[item] = map.GetValueOrDefault(item) + size;
        }
    }
}

var list = map.Values.ToList();
int score = 0;


foreach (var item in list)
{
    if(item <= 100000)
    {
        score += item;
    }
   
}


//Part 2
int remainingDiskSpace = 70000000 - list[0];
int DiskSpaceToRemove = 30000000 - remainingDiskSpace;
List<int> removes = new List<int>();
foreach (var item in list)
{
    if(item > DiskSpaceToRemove)
    {
        removes.Add(item);
    }
}
int dictToRemove = 0;
for (int i = 0; i < removes.Count - 1; i++)
{
    if (removes[i] < removes[i+1])
    {
        dictToRemove = removes[i];
    }
}
Console.WriteLine("(Part1): " + score); //part1 Result: 1306611
Console.WriteLine("(Part2): "+ dictToRemove); //part2 Result: 13210366

File: 2022/07/part1.js
const fs = require('fs');
const data = fs.readFileSync("input.txt", {encoding: "utf-8"}).replace(/\r/g, "").trim().split("\n");

let path = [];
let map = {};

let x = data;

for (let i = 0; i < x.length; i++)
{  
    var h = x[i].split(' ');  
    if (h[0] == "$")
    {
        if (h[1] == "cd")
        {
            if (h[2] == "..")
            {
                path.pop();
            }
            else if (h[2] == "/")
            {
                path.push(path.join('',path) + h[2]);
            }
            else
            {               
                path.push(path.join('',path) + h[2]);
            }
        }
    } else if (h[0] != "dir") {
        var size = Number(h[0]);
        path.forEach(element => {
            if(map[element] == undefined) {
                map[element] = size;
            } else {
                map[element] = map[element] + size
            }           
        });
    }
}
let score = 0;
for(var key in map) {
    var value = map[key];
    if(value <= 100000)
    {
        score += value;
    }
  }
console.log("Result: " + score); //1306611
